.github/workflows/haskell.yml

```diff
@@ -46,9 +51,12 @@ jobs:
         os: [ubuntu-latest]
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
+
     - name: Set NIGHTLY environment variable if the job was triggered by the scheduler
-      if: ${{ github.event_name == 'schedule' }}
+      if: "${{ github.event_name == 'schedule'
+            || contains(github.event.pull_request.title, 'nightly')
```

Comment:
This line is just to enable me to test that the fixes work. I'll take it out before we merge.

In the future, we'll probably use the manual triggering for this, but using it for this PR would mean I have to build every iteration twice.

Reply:
> I'll take it out before we merge.

We don't have to take it out. I think it is reasonable to always run nightly tests if a PR has nightly in title.